feat(allocator): add `Allocator::end_ptr` method

overlookmotel opened the pull request:
Add a method `Allocator::end_ptr` which returns a pointer to the end of the current chunk of an `Allocator`, after the `ChunkFooter`. For `Allocator`s used for raw transfer, this pointer will point to the `RawTransferMetadata` which is stored after the allocator chunk.

camc314 commented:
### Merge activity

* **Jul 18, 8:47 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
feat(allocator): add `Allocator::end_ptr` method (#12330)

Add a method `Allocator::end_ptr` which returns a pointer to the end of the current chunk of an `Allocator`, after the `ChunkFooter`. For `Allocator`s used for raw transfer, this pointer will point to the `RawTransferMetadata` which is stored after the allocator chunk.
